src/lighting/TrainLighting.cpp

```diff
@@ -0,0 +1,307 @@
+#include "TrainLighting.hpp"
+
+namespace godot {
+    const char *TrainLighting::WRAPPER_POSITION_CHANGED_SIGNAL = "wrapper_position_changed";
+
+    void TrainLighting::_bind_methods() {
+        ClassDB::bind_method(D_METHOD("set_head_light_color", "color"), &TrainLighting::set_head_light_color);
+        ClassDB::bind_method(D_METHOD("get_head_light_color"), &TrainLighting::get_head_light_color);
+        ADD_PROPERTY(PropertyInfo(Variant::COLOR, "head_light/color"), "set_head_light_color", "get_head_light_color");
+        ClassDB::bind_method(D_METHOD("set_head_light_dimming_multiplier", "multiplier"), &TrainLighting::set_dimming_multiplier);
+        ClassDB::bind_method(D_METHOD("get_head_light_dimming_multiplier"), &TrainLighting::get_dimming_multiplier);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "head_light/dimming_multiplier"), "set_head_light_dimming_multiplier", "get_head_light_dimming_multiplier");
+        ClassDB::bind_method(D_METHOD("set_head_light_normal_multiplier", "multiplier"), &TrainLighting::set_normal_multiplier);
+        ClassDB::bind_method(D_METHOD("get_head_light_normal_multiplier"), &TrainLighting::get_normal_multiplier);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "head_light/normal_multiplier"), "set_head_light_normal_multiplier", "get_head_light_normal_multiplier");
+        ClassDB::bind_method(D_METHOD("set_high_beam_dimmed_multiplier", "multiplier"), &TrainLighting::set_high_beam_dimmed_multiplier);
+        ClassDB::bind_method(D_METHOD("get_high_beam_dimmed_multiplier"), &TrainLighting::get_high_beam_dimmed_multiplier);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "head_light/high_beam/dimming_multiplier"), "set_high_beam_dimmed_multiplier", "get_high_beam_dimmed_multiplier");
+        ClassDB::bind_method(D_METHOD("set_high_beam_multiplier", "multiplier"), &TrainLighting::set_high_beam_multiplier);
+        ClassDB::bind_method(D_METHOD("get_high_beam_multiplier"), &TrainLighting::get_high_beam_multiplier);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "head_light/high_beam/normal_multiplier"), "set_high_beam_multiplier", "get_high_beam_multiplier");
+        ClassDB::bind_method(
+                D_METHOD("set_default_selector_position", "position"), &TrainLighting::set_default_selector_position);
+        ClassDB::bind_method(D_METHOD("get_default_selector_position"), &TrainLighting::get_default_selector_position);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::INT, "lights/default_position"),
```

Comment:
Little inconsistency between setget and property name. Consider set/get_lights_default_position